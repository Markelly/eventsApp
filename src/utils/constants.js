export const IP_URL = "http://ip-api.com/json";
export const DISCOVER_URL = "https://app.ticketmaster.com/discovery/v2";
export const API_KEY = "4diZ7tJQArO9sxHiVWMASl5udVTAoI78";

export const EVENTS_PATH = "/events";
export const FAVORITES_PATH = "/favorites";

export const SORT_OPTIONS = [
  { text: "Upcoming events", value: "date,asc" },
  { text: "Name", value: "name,asc" },
  { text: "Relevance", value: "relevance,asc" },
  { text: "Distance", value: "distance,asc" }
];

export const monthName = [
  "January",
  "February",
  "March",
  "April",
  "May",
  "June",
  "July",
  "August",
  "September",
  "October",
  "November",
  "December"
];

export const weekDayName = [
  "Sunday",
  "Monday",
  "Tuesday",
  "Wednesday",
  "Thursday",
  "Friday",
  "Saturday"
];
