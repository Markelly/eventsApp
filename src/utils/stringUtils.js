import { EVENTS_PATH } from "./constants";

export const getEventsList = eventsList => {
  let event = {};
  let events = [];
  eventsList.forEach(item => {
    event.name = item.name;
    event.date = item.dates;
    event.price = getPrice(item);
    event.id = item.id;
    event.favorite = false;
    event.image = item.images[0].url;
    event.url = item.url;
    events.push(event);
    event = {};
  });
  return events;
};

export const getPrice = item => {
  const min = item.priceRanges ? item.priceRanges[0].min : 0;
  const max = item.priceRanges ? item.priceRanges[0].max : 0;
  return item.priceRanges ? `€${min} - €${max}` : "Free";
};

export const isEventsPage = path => {
  return path === EVENTS_PATH;
};

export const getImageContext = icon => {
  const images = require.context("../../public/images", false, /\.png$/);
  return images(`./${icon}.png`);
};

export default {
  getEventsList,
  getPrice,
  isEventsPage,
  getImageContext
};
