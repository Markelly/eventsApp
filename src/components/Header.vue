<template>
  <div class="header">
    <div class="header__menu">
      <button
        v-if="isEventsRoute"
        class="header__menu__item"
        @click="goToFavorites"
        data-favorites-button
      >
        My favorites
      </button>
      <button
        v-else
        class="header__menu__item"
        @click="goToEvents"
        data-events-button
      >
        See all events
      </button>
    </div>
  </div>
</template>

<script>
import { EVENTS_PATH, FAVORITES_PATH } from "../utils/constants";
import { isEventsPage } from "../utils/stringUtils";

export default {
  name: "Header",
  methods: {
    goToEvents() {
      this.$router.push(EVENTS_PATH);
    },
    goToFavorites() {
      this.$router.push(FAVORITES_PATH);
    }
  },
  computed: {
    isEventsRoute() {
      return isEventsPage(this.$route.path);
    }
  }
};
</script>

<style scoped lang="scss">
@import "../styles/events-app";
.header {
  &__menu {
    display: flex;

    &__item {
      padding: 8px 16px;
      font-size: 24px;
      border: solid 2px;
      border-radius: 25px;
      color: $color1;
      background-color: $color4;

      &:focus {
        outline:0;
      }

      &:hover {
        color: $color4;
        border-color: $color4;
        background-color: $color1;
      }
    }
  }
}
</style>
