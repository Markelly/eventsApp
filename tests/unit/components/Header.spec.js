import Header from "../../../src/components/Header";
import { createLocalVue, shallowMount } from "@vue/test-utils";
import Router from "vue-router";
import { EVENTS_PATH, FAVORITES_PATH } from "../../../src/utils/constants";

describe("Header", () => {
  const localVue = createLocalVue();
  localVue.use(Router);
  let wrapper, mockRoute, spyRoutePush;

  beforeEach(() => {
    mockRoute = new Router({
      routes: [
        {
          path: EVENTS_PATH
        },
        {
          path: FAVORITES_PATH
        }
      ]
    });
    spyRoutePush = jest.spyOn(mockRoute, "push");
    mockRoute.push(EVENTS_PATH);
    wrapper = shallowMount(Header, {
      localVue,
      router: mockRoute,
      computed: {
        isEventRoute() {
          return true;
        }
      }
    });
  });

  test("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should render events page", () => {
    expect(wrapper.find("[data-favorites-button]")).toBeTruthy();
  });

  test("should render favorites page", () => {
    mockRoute.push(FAVORITES_PATH);
    expect(wrapper.find("[data-suggest-events]")).toBeTruthy();
  });

  test("should go to events page", () => {
    wrapper.vm.goToEvents();
    expect(spyRoutePush).toHaveBeenCalledWith(EVENTS_PATH);
  });

  test("should go to favorites page", () => {
    wrapper.vm.goToFavorites();
    expect(spyRoutePush).toHaveBeenCalledWith(FAVORITES_PATH);
  });
});
