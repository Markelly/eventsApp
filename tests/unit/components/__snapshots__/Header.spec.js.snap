// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> should render the component 1`] = `
<div class="header">
  <div class="header__menu"><button data-favorites-button="" class="header__menu__item">
      My favorites
    </button></div>
</div>
`;
